# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../protothread/makefile.patch
# Copyright (C) 2009 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./Makefile.orig	2009-01-18 19:12:34.000000000 +0000
+++ ./Makefile	2009-01-18 19:40:18.000000000 +0000
@@ -1,15 +1,36 @@
+CC = gcc
+AR = ar
 CFLAGS = -g -Wall
 
+prefix = /usr/local
+bindir = $(prefix)/bin
+libdir = $(prefix)/lib
+DESTDIR=
+
+TARGETS = pttest protothread.a
+
 OBJS = \
        protothread_test.o \
        protothread_lock.o \
        protothread_sem.o \
        protothread.o \
 
+
+.PHONY: all clean install
+
+all: $(TARGETS)
+
 pttest:	$(OBJS) protothread.h
 	$(CC) $(CFLAGS) -o $@ $(OBJS)
 
+protothread.a: $(OBJS)
+	$(AR) rcs $@ $(OBJS)
+
 $(OBJS): protothread.h
 
 clean:
-	rm -f *.o
+	rm -f $(OBJS) $(TARGETS)
+
+install:
+	cp -vf pttest $(DESTDIR)$(bindir)/
+	cp -vf protothread.a $(DESTDIR)$(libdir)/
